subworkflows/local/deepvariant_error_polishing/main.nf

```diff
@@ -0,0 +1,204 @@
+        uniq_assembly_ch
+    )
+
+    // map reads with pbmm2 to complete assemblies (chunks are not used in that step)
+    DVPOLISH_PBMM2_ALIGN (
+        input.assembly_ch,
+        input.reads_ch
+    )
+
+    def path_closure = {meta, files -> files.collect(){[meta, it ]}}
+
+    DVPOLISH_PBMM2_ALIGN.out.bam
+    .flatMap(path_closure)
+    .combine(DVPOLISH_CHUNKFA.out.bed.flatMap(path_closure), by:0)
+    .map { map, bam, bed -> [ map + [mergeID: bed.baseName], bam, bed]
+    }
+    .set { bam_bed_ch }
+
+    DVPOLISH_PBMM2_ALIGN.out.bai
+    .flatMap(path_closure)
+    .combine(DVPOLISH_CHUNKFA.out.bed.flatMap(path_closure), by:0)
+    .map { meta, bai, bed -> bai }
+    .set { bai_ch }
+    // split bam files according to bed file chunks 
+    SAMTOOLS_VIEW (bam_bed_ch, [[],[]], bai_ch)
+
+    // index the splitted bam files 
+    SAMTOOLS_INDEX_FILTER(SAMTOOLS_VIEW.out.bam)
+
+    SAMTOOLS_VIEW.out.bam
+    .groupTuple(by:0)
+    .branch { meta, bam_list ->
+        merge: bam_list.size() > 1
+        link: true
+    }
+    .set { bam_merge_ch }
+
+    // in case multiple reads files are present, all corresping bam files 
+    // the where splitted in the previous step need to be merged. key:bed file ID
+    SAMTOOLS_MERGE(
+        bam_merge_ch.merge,
+        [[],[]],
+        [[],[]]
+    )
+    // index merged bam files
+    SAMTOOLS_INDEX_MERGE(SAMTOOLS_MERGE.out.bam)
+
+    bam_merge_ch.link
+    .map { meta, bam -> [ meta, *bam ]} // the spread operator (*) flattens the bam list
+    .join(SAMTOOLS_INDEX_FILTER.out.bai, by:0)
+    .mix(SAMTOOLS_MERGE.out.bam
+        .join(SAMTOOLS_INDEX_MERGE.out.bai, by:0)
+    )
+    .join(bam_bed_ch
+    .map { meta, bam, bed -> [meta, bed]}
+    .unique())
+    .set {deepvariant_ch}
+    // run deepvariant and the chunked bam files 
+    DEEPVARIANT(
+        deepvariant_ch,
+        uniq_assembly_ch,
+        SAMTOOLS_FAIDX.out.fai,
+        [[],[]]     // tuple val(meta4), path(gzi)
+    )
+
+    DEEPVARIANT.out.vcf
+    .join(DEEPVARIANT.out.vcf_tbi, by:0)
+    .set { bcftools_view_ch }
+    // filter vcf files for PASS and homozygous varinats
+    // TODO add a minimim and maximum coverage filter ??? Needs to be tested
+    BCFTOOLS_VIEW (
+        bcftools_view_ch,
+        [], // path(regions)
+        [], // path(targets)
+        [] // path(samples)
+    )
+
+    // index vcf file 
+    TABIX_TABIX(
+        BCFTOOLS_VIEW.out.vcf
+    )
+
+    // in case of multiple vcf files, merge them prior the consenus step
+    BCFTOOLS_VIEW.out.vcf
+    .map { meta, vcf -> [ meta - meta.subMap('mergeID'), vcf ] }
+    .groupTuple(by:0)
+    .set { filt_vcf_list_ch }
+
+    TABIX_TABIX.out.tbi
+    .map { meta, tbi -> [ meta - meta.subMap('mergeID'), tbi ] }
+    .groupTuple(by:0)
+    .set { filt_tbi_list_ch }
+
+    filt_vcf_list_ch
+    .join(filt_tbi_list_ch, by:0)
+    .branch { meta, vcf_list, vcf_index_list ->
+        merge: vcf_list.size() > 1
+        other: true
```

Comment:
Maybe rename this channel to `singleton`, and then the other to `multiples`.
```suggestion
        multiples: vcf_list.size() > 1
        singleton: true
```
Same with the other one named `link` above.